src/decode-pppoe.c

```diff
@@ -185,46 +198,45 @@ int DecodePPPOESession(ThreadVars *tv, DecodeThreadVars *dtv, Packet *p,
 
             case PPP_VJ_UCOMP:
 
-                if(len < (PPPOE_SESSION_HEADER_LEN + IPV4_HEADER_LEN))    {
+                if (len - pppoesh_len < IPV4_HEADER_LEN) {
```

Comment:
I think this `len - pppoesh_len` can underflow. `len` is validated to `>= PPPOE_SESSION_HEADER_MIN_LEN`, so it can be `PPPOE_SESSION_HEADER_MIN_LEN`. `pppoe_len` can be `PPPOE_SESSION_HEADER_MIN_LEN + 1` in the double octet variant.

I'm not sure what the result is here, as we have a `uint32_t`, `uint8_t` and `int` in this single check, but I think the code shouldn't force me to question this.